src/generator.cpp

```diff
@@ -1222,18 +1247,42 @@ void Generator::GeneratorImpl::processEquationAst(const GeneratorEquationAstPtr
 
                 mGenerator->addError(err);
             } else {
-                mVoi = variable;
+                // We have found our variable of integration, but this may not
+                // be the one defined in our first component (i.e. the component
+                // under which we are likely to expect to see the variable of
+                // integration to be defined), so go through our components and
+                // look for the first occurrence of our variable of integration.
+
+                ModelPtr model = owningModel(std::dynamic_pointer_cast<Component>(variable->parent()));
```

Comment:
Why are you dynamic casting the returned result?

Reply:
Clearly an oversight at my end. So, yes, well spotted. Consider it fixed.

Reply:
Fixed in commit df08fd8.